File: appkit-creating-and-customizing-the-touch-bar/Objective-C/NSTouchBar Catalog/BackgroundWindow/BackgroundViewController.m
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
View controller for the background of the Background Window test.
*/

#import "BackgroundViewController.h"

@interface BackgroundViewController ()

// Accessory view controller to hold the Set Background button as part of the window's title bar.
@property (nonatomic, strong) NSTitlebarAccessoryViewController *titlebarAccessoryViewController;

@end


#pragma mark -

@implementation BackgroundViewController

- (void)viewWillAppear
{
    [super viewWillAppear];
    
    // Set the window's title bar accessory view controller that holds the Set Background button.
    _titlebarAccessoryViewController = [self.storyboard instantiateControllerWithIdentifier:@"ChangeBackground"];
    self.titlebarAccessoryViewController.layoutAttribute = NSLayoutAttributeBottom;
    [self.view.window addTitlebarAccessoryViewController:self.titlebarAccessoryViewController];
}

@end

File: appkit-creating-and-customizing-the-touch-bar/Swift/NSTouchBar Catalog/BackgroundViewController.swift
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
The view controller for the Background Window test case.
*/

import Cocoa

class BackgroundViewController: NSViewController {
    
    @IBOutlet weak var imageView: NSImageView!
    
    // Accessory view controller to hold the Set Background button as part of the window's title bar.
    var titlebarAccessoryViewController: NSTitlebarAccessoryViewController!
    
    override func viewWillAppear() {
        super.viewWillAppear()
        
        titlebarAccessoryViewController =
            storyboard?.instantiateController(withIdentifier: NSStoryboard.Name("ChangeBackground")) as? NSTitlebarAccessoryViewController
        titlebarAccessoryViewController.layoutAttribute = .bottom
        self.view.window?.addTitlebarAccessoryViewController(self.titlebarAccessoryViewController)
    }
}
